### PR TITLE
Adding support for Windstudio's Wind X R1 keyboard

### PR DESCRIPTION
**Pull request by ChrisGVE**
The change includes the keyboard layout, data driven with .json file, and a default layout as a starting point to customize a board.

## Description

The change includes a folder for all Windstudio keyboards (only one is provided in this change) and a subfolder for the Wind X R1 keyboard, it is organized as wind_x / r1 such that the existing r2 could be added later.

The keyboard is fully configured via a keyboard.json file which presents the 2 main layouts that this keyboard offers. In addition a default layout is provided.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/windstudio/wind_x/r1/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    "keyboard_name": "Wind X R1",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": false,
+        "rgblight": false
```
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/keyboard.json:14]:
```suggestion
        "extrakey": true,
        "mousekey": true,
        "nkro": true
```

File: keyboards/windstudio/wind_x/r1/keyboard.json
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/keyboard.json:1]: Please run `qmk format-json -i` on this file.

File: keyboards/windstudio/wind_x/r1/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+            "B3"
+        ],
+        "rows": ["F0", "F1", "D2", "D1", "D0"]
+    },
+    "diode_direction": "COL2ROW",
+    "layout_aliases": {
+        "LAYOUT_tsangan": "LAYOUT_7u_space_bar_split_bs"
+    },
+    "layouts": {
+        "LAYOUT_default": {
```
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/keyboard.json:51]:
```suggestion
        "LAYOUT": {
```

File: keyboards/windstudio/wind_x/r1/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		KC_LCTL, KC_LGUI, KC_LALT,                   LT(1, KC_SPC),                      KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT,             KC_P0,   KC_PDOT
+    ),
+
+	[1] = LAYOUT_default(
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,   _______,   _______, _______, _______, _______,
+		QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,   _______,   _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,            _______,   _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                       _______, _______, _______, _______,
+		_______, _______, _______,                   _______,                            _______, _______, _______, _______, _______,              _______, _______
+    ),
```
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/keymaps/default/keymap.c:36]:
```suggestion
	[0] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,   KC_DEL,   KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,   KC_PGUP,  KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,             KC_PGDN,  KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_LCTL, KC_LGUI, KC_LALT,                   LT(1, KC_SPC),                      KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT,             KC_P0,   KC_PDOT
    ),

	[1] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,   _______,   _______, _______, _______, _______,
        QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,   _______,   _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,            _______,   _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                       _______, _______, _______, _______,
        _______, _______, _______,                   _______,                            _______, _______, _______, _______, _______,              _______, _______
    )
```

File: keyboards/windstudio/wind_x/r1/readme.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# WindX R1 by Windstudio
+
+![Windstudio](https://windstudio.store)
```
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/readme.md:3]: This doesn't appear to be an image.

File: keyboards/windstudio/wind_x/r1/readme.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+A large layout keyboard without the function keys. [More info on geekhack.org](https://geekhack.org/index.php?topic=114767.0)
+
+-   Keyboard Maintainer: [Christian C. Berclaz](https://github.com/chrisgve)
+-   Hardware Supported: Wind X R1
```
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/readme.md:8]:
```suggestion
* Keyboard Maintainer: [Christian C. Berclaz](https://github.com/chrisgve)
* Hardware Supported: Wind X R1 w/ ATmega32U4 microcontroller
* Hardware Availability: [windstudio.store](https://windstudio.store/collections/wind-x)
```

File: keyboards/windstudio/wind_x/r1/keymaps/default/rules.mk
Comment (waffle87) [inline keyboards/windstudio/wind_x/r1/keymaps/default/rules.mk:1]: This file should be removed.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/windstudio/wind_x/r1/readme.md (hunk 4 above)
Comment (ChrisGVE) [reply keyboards/windstudio/wind_x/r1/readme.md:3]: Question here: should it be the logo of the studio or an image of the keyboard?

File: keyboards/windstudio/wind_x/r1/keyboard.json
Comment (ChrisGVE) [reply keyboards/windstudio/wind_x/r1/keyboard.json:1]: Done

File: keyboards/windstudio/wind_x/r1/keymaps/default/rules.mk
Comment (ChrisGVE) [reply keyboards/windstudio/wind_x/r1/keymaps/default/rules.mk:1]: Done

File: keyboards/windstudio/wind_x/r1/readme.md (hunk 4 above)
Comment (ChrisGVE) [reply keyboards/windstudio/wind_x/r1/readme.md]: In the meantime I have linked the studio logo, obviously I can change as required.

Comment (ChrisGVE) [reply keyboards/windstudio/wind_x/r1/readme.md]: Checked other keyboards, I've linked a picture of the keyboard.

**Review by waffle87 — APPROVED** [review]

**Pull request merged**